src/dune_engine/action_exec.ml

```diff
@@ -542,6 +542,34 @@ type input =
   ; action : Action.t
   }
 
+module Install_root = struct
+  let path_sep = if Sys.win32 then ';' else ':'
+
+  let ( ^/ ) = Filename.concat
+
+  let exe = if Sys.win32 then ".exe" else ""
+
+  let best_prog dir prog =
+    let fn = dir ^/ prog ^ ".opt" ^ exe in
+    if Sys.file_exists fn then Some fn
+    else
+      let fn = dir ^/ prog ^ exe in
+      if Sys.file_exists fn then Some fn else None
+
+  let find_ocaml_prog path prog =
+    List.find_map path ~f:(fun dir -> best_prog dir prog)
+
+  let get (env : Env.t) =
+    match Env.get env "OPAM_SWITCH_PREFIX" with
+    | Some dir -> Some dir
+    | None ->
+      let open Option.O in
+      let* path_string = Env.get env "PATH" in
+      let path = String.split path_string ~on:path_sep in
+      let* ocamlc = find_ocaml_prog path "ocamlc" in
```

Comment:
this can be let+ letting you get rid of the Some

Reply:
done.